data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/actions/CombineAggregateAction.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate.actions;
+
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.plugins.processor.aggregate.AggregateAction;
+import com.amazon.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class CombineAggregateAction implements AggregateAction {
+    static final String COMBINED_EVENT_TYPE = "combined_event";
+
+    @Override
+    public AggregateActionResponse handleEvent(Event event, Map<Object, Object> groupState) {
+        if (groupState != null)
```

Comment:
The `groupState` should be guaranteed non-null by the `AggregateProcessor`. This check should be unnecessary.

Reply:
So for the case where the groupState doesn't exist for a certain group yet, will the `AggregateProcessor` just not call handleEvent?

Reply:
Or should this null check be replaced with something like `if (groupState.size() == 0)`

Reply:
The AggregateProcessor should create a new `Map` when the groupState is created.